troposphere/haze/source/usb_session.cpp

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <haze.hpp>
+
+namespace haze {
+
+    namespace {
+
+        static constexpr u32 DefaultInterfaceNumber = 0;
+
+    }
+
+    Result UsbSession::Initialize1x(const UsbCommsInterfaceInfo *info) {
+        struct usb_interface_descriptor interface_descriptor = {
```

Comment:
Should all these structs be const? Also, vertically aligned = for all of these.

Reply:
Can't be const, as they are passed into a method which requires a non-const pointer (usbDsInterface_GetDsEndpoint).